### DISCUSSION TITLE
C# API Docs Beta? Source Code? Extra markdown syntax support?

### DISCUSSION THREAD

**Discussion by ApolonTorq** [retypeapp/retype#31] [Q&A] [answered]
Is it possible to participate in the C# API docs generation beta?

Is there a public repo for the source code for retypeapp? Or is it the intent that retypeapp is a commercial closed source product?

Can retypeapp's markdown processing be extended? e.g. support for mermaid and maybe some more esoteric markdown syntax?

**Comment by geoffreymcgill** [comment, marked as answer]
> Is it possible to participate in the C# API docs generation beta?

Yes, there is limited C# API reference doc generation support built into Retype `v1.6.0`. 

Add the following `api` config to your project `retype.json`.

```json
{    
  "api": {
    "projects": [
      {
        "input": "../path-to-your-csharp-project"
      }
    ]
  }
}
```

At the moment, the only other supported config is `output`, which controls which folder within the [`output`](https://retype.com/configuration/project/#output) that Retype will send the built API docs.

You can see the `API Reference` docs in action here:

https://docs.mojee.io

Here's the `api` config for the Mojee API docs:

https://github.com/MojeeIO/Mojee/blob/master/retype.json#L16-L22

```json
    "api": {
      "projects": [
        {
          "input": "../mojee-dev/src/csharp/Mojee"
        }
      ]
    },
```

For the Mojee API docs, we also add the following `index.yml` to configure the folder `icon` and `label`. Both those configs will be possible to set in the `api` config of `retype.json` as of `v1.7.0`. 

https://github.com/MojeeIO/Mojee/blob/master/api/index.yml

> Is there a public repo for the source code for retypeapp? Or is it the intent that retypeapp is a commercial closed source product?

Retype is closed source and I don't think we'll be open-sourcing the project. We've had some big open-source projects in the past and the commonly perceived advantages from open-sourcing never really materialized.

We're going to focus on trying to find a balance to ensure 90%+ of Retype projects (personal, open-source, education, non-profit, and commercial) are covered by a free $0 license. There will be features that deliver a boat load of value and unlocking that value will require a `pro` license. There is a full time dedicated professional team building Retype and it takes real money to keep this machine running. 

> Can retypeapp's markdown processing be extended? e.g. support for mermaid and maybe some more esoteric markdown syntax?

Not yet, but it is something we have discussed and will be exploring options. We can certainly investigate Mermaid. After an initial quick review, it doesn't look much different than our Prism integration for code block syntax highlighting. 

I created a new Feature request issue to track our progress on supporting Mermaid, see https://github.com/retypeapp/retype/issues/33. 

Performance and stability are primary goals, both during generation time and runtime, so we need to tread carefully if anything compromises those principles. Prism does impact the client runtime performance of Retype (IMHO), so I'd love to build Prims out of the project with another solution we've been thinking about.

Hope this helps answer your questions.

**Reply by ApolonTorq** [reply]
Is it possible to get rid of this limitation?
> WARNING: This edition of Retype is limited to generating one API Project reference only.

Or is it too early in the development process just yet?

**Reply by geoffreymcgill** [reply]
> Is it possible to get rid of this limitation?

Yes, it is possible. Please email hello@object.net with the request.
